server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -205,6 +217,38 @@ public void onFailure(Exception e) {
             });
     }
 
+    /**
+     * Retrieves all synonym rules for a synonym set.
+     *
+     * @param synonymSetId
+     * @param listener
+     */
+    public void getSynonymSetRules(String synonymSetId, ActionListener<PagedResult<SynonymRule>> listener) {
+        // Check the number of synonym sets, and issue a warning in case there are more than the maximum allowed
+        client.prepareSearch(SYNONYMS_ALIAS_NAME)
+            .setSource(new SearchSourceBuilder().size(0).trackTotalHits(true))
+            .execute(listener.delegateFailureAndWrap((searchListener, countResponse) -> {
+                long totalSynonymRules = countResponse.getHits().getTotalHits().value;
+                if (totalSynonymRules > maxSynonymsSets) {
+                    logger.warn(
```

Comment:
I wonder if this should be logged as an error?

Reply:
It's not an error condition per se IMO - just inconsistency on applying synonyms. It will be functionally wrong, but I don't think it's something that will imply a general malfunction.

Happy to change if you think this should be brought to attention as error.

Reply:
that works for me thanks